00XX-common-ledger-protocol/00XX-common-ledger-protocol.md

```diff
@@ -0,0 +1,273 @@
+  +---------------+
+1 | Request ID    |
+2 | (2)           |
+3 |               |
+4 |               |
+  +---------------+
+1 | Length Prefix |
+2 | (3)           |
+  +---------------+
+  | Packet-       |
+  | Specific      |
+  | Data          |
+  . (4)           |
+  .               |
+  .               |
+  |               |
+  +---------------+
+1 | Side Protocol |
+2 | Count (5)     |
+  +---------------+
+  | Side Protocol |
+  | Data          |
+  . (6)           |
+  .               |
+  .               |
+  |               |
+  +---------------+
+```
+
+1. **Type**: A 1-byte value describing what type of CLP packet this is.
+The [CLP ASN.1 spec] describes what the different type IDs mean.
+
+2. **Request ID**: A 4-byte value used to correlate this requests and
+responses.
+
+3. **Length Prefix**: A 1-byte (if under 128) or 2-byte value, containing the
+combined length of the packet-specific data and side protocol data sections.
+
+4. **Packet-Specific Data:** Fields specific to the type of CLP packet. Variable
+length.
+
+5. **Side Protocol Count:** Variable-length integer containing the number of
+side protocols carried by this packet.
+
+5. **Side Protocol Data:** A list of key-value pairs, where the key is a
+length-prefixed ascii string, and the value is a length-prefixed octet string.
+The key signifies the side protocol's name, and the value is the data.
+
+## Flow
+
+CLP uses a simple RPC flow. A request-type CLP packet is sent, and a
+response-type CLP packet is sent in response with the same request ID.
+
+If a malformed CLP packet is sent, there should be no response. If a
+response-type CLP packet is sent without matching a previous request, it should
+be ignored.  If a request has already been responded to and another response is
+sent, the second response should be ignored.
+
+Because it would be too slow to atomically save all requestIds that are
+processed, they are not idempotent. It is the responsibility of the requestor
+to make sure they don't duplicate requestIds. The implementation should ensure
+that no two in-flight requests are sent out with the same requestId. The
+responder should always send back a response to a request with the same
+requestId.
+
+The request types are `Message`, `Prepare`, `Fulfill` and `Reject`, and the
+response types are `Ack` (which may be [removed]), `Response`, and `Error`.
+
+### Message
+
+`Message` is used for sending information to the peer. It contains no message
+data, only side protocol data. ILQP packets are attached under the protocol
+`ilp`.
+
+#### Response
+
+- `Response` is returned if the peer acknowledges the `Message`. If the peer
+  has data to send in reply (e.g. a quote response), it is carried in the side
+protocol data.
+
+- `Error` is returned if the peer is not able to process the message, or there
+  was an unexpected error and further messages should not be sent.
+
+### Prepare
+
+`Prepare` is used to create a transfer on the bilateral ledger. The message
```

Comment:
Can we call this PrepareTransfer?

Verbosity in descriptions is really helpful to anyone reading this for the first time and it adds nothing to the encoded data.